akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -671,20 +591,27 @@ private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Mo
   protected def mergeAttributes(parent: Attributes, current: Attributes): Attributes =
     parent and current
 
+  private val matValSrc: ju.Map[MaterializedValueNode, List[MaterializedValueSource[Any]]] = new ju.HashMap
+  def registerSrc(ms: MaterializedValueSource[Any]): Unit = {
+    if (MaterializerSession.Debug) println(s"registering source $ms")
+    matValSrc.get(ms.computation) match {
+      case null ⇒ matValSrc.put(ms.computation, ms :: Nil)
```

Comment:
are these kind of `match` statements properly optimized? I.e. is it equivalent to an `if`?